### PR TITLE
Frontend server: add feature flag for kubeflow deployment

### PR DESCRIPTION
**Pull request by Bobgy**
/assign @gaoning777

Pipeline UI pod will pass this feature flag to client: "DEPLOYMENT=KUBEFLOW".
In browser UI code, the flag can be accessed as `window.KFP_FLAGS.DEPLOYMENT`.

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/2631)

### CODE REVIEW

File: frontend/public/index.html [hunk 1]
```diff
@@ -23,6 +23,10 @@
     <link rel="manifest" href="%PUBLIC_URL%/static/manifest.json">
     <link rel="shortcut icon" href="%PUBLIC_URL%/static/favicon.ico">
     <title>Kubeflow Pipelines</title>
+    <script>
+      window.KFP_FLAGS={};
+      window.KFP_FLAGS.MULTI_USER=0;
```
Comment (gaoning777) [inline frontend/public/index.html]: Does this work for common browsers such as chrome, firefox, safari?

Comment (Bobgy) [reply frontend/public/index.html]: Yes, this is very standard html.

File: frontend/server/server.ts [hunk 2]
```diff
@@ -399,13 +409,50 @@ app.all(BASEPATH  + '/' + v1beta1Prefix + '/*', proxy({
   target: apiServerAddress,
 }));
 
+const DEFAULT_FLAG = 'window.KFP_FLAGS.DEPLOYMENT=null';
+function modifyFeatureFlags(indexHtml: string): string {
+  if (DEPLOYMENT === Deployments.KUBEFLOW) {
+    return indexHtml.replace(DEFAULT_FLAG, 'window.KFP_FLAGS.MULTI_USER="KUBEFLOW"');
```
Comment (gaoning777) [inline frontend/server/server.ts]: Should this also be window.KFP_FLAGS.DEPLOYMENT?

Comment (Bobgy) [reply frontend/server/server.ts]: Oops, thanks for the catch!

**Pull request merged**